workos/api_keys.py

```diff
@@ -0,0 +1,54 @@
+from typing import Protocol
+
+from workos.types.api_keys import ApiKey
+from workos.typing.sync_or_async import SyncOrAsync
+from workos.utils.http_client import AsyncHTTPClient, SyncHTTPClient
+from workos.utils.request_helper import REQUEST_METHOD_POST
+
+API_KEY_VALIDATION_PATH = "api_keys/validations"
+
+
+class ApiKeysModule(Protocol):
+    def validate_api_key(self, *, value: str) -> SyncOrAsync[ApiKey]:
+        """Validates the configured API key.
+
+        Returns:
+            ApiKey: The validated API key details containing
+                information about the key's name and usage
+
+        Raises:
+            AuthenticationException: If the API key is invalid or
+                unauthorized (401)
+            NotFoundException: If the API key is not found (404)
+            ServerException: If the API server encounters an error
+                (5xx)
+        """
+        ...
+
+
+class ApiKeys(ApiKeysModule):
+    _http_client: SyncHTTPClient
+
+    def __init__(self, http_client: SyncHTTPClient):
+        self._http_client = http_client
+
+    def validate_api_key(self, *, value: str) -> ApiKey:
+        response = self._http_client.request(
+            API_KEY_VALIDATION_PATH, method=REQUEST_METHOD_POST, json={
+                "value": value}
+        )
+        return ApiKey.model_validate(response["api_key"])
```

Comment:
When I passed an invalid value into this method, I saw this error message:

> Input should be a valid dictionary or instance of ApiKey

From [the docs](https://workos.com/docs/reference/authkit/api-keys/validate):

> If the key is invalid, the endpoint returns null for the api_key field.

Do we need to handle the invalid case differently?